server.py

```diff
@@ -330,10 +330,39 @@ async def get_current_crypto_price(ticker: str) -> str:
     return json.dumps(snapshot, indent=2)
 
 
+@mcp.tool()
+async def get_filings(
+    ticker: str, 
+    filing_type: str
+) -> str:
+    """Get the current / latest price of a crypto currency.
+
+    Args:
+        ticker: Ticker symbol of the company (e.g. AAPL, GOOGL)
+        filing_type: Type of the filing (10-K, 10-Q, 8-K, 4, 144)
+    """
+    # Fetch data from the API
+    url = f"{FINANCIAL_DATASETS_API_BASE}/filings?ticker={ticker}&filing_type={filing_type}"
+    data = await make_request(url)
+
+    # Check if data is found and contains filings
+    if not data or not (filings := data.get("filings", [])):
+        return "Unable to fetch filings or no filings found."
+
+    # Stringify the current price
+    return json.dumps(filings, indent=2)
+
+
 if __name__ == "__main__":
     # Log server startup
     logger.info("Starting Financial Datasets MCP Server...")
 
+    # Warn if no API key found
+    if not os.environ.get("FINANCIAL_DATASETS_API_KEY"):
+        logger.warning(
```

Comment:
Helpful to debug if the API key is actually being loaded.

![Screenshot from 2025-05-06 14-07-54](https://github.com/user-attachments/assets/eab099b7-ae13-41fd-9d1a-121e1c979e51)

vs.

![Screenshot from 2025-05-06 14-08-15](https://github.com/user-attachments/assets/e9695950-6243-48b2-b36f-521939a47197)